Sources/TuistKit/Cache/CacheController.swift

```diff
@@ -108,15 +108,56 @@ final class CacheController: CacheControlling {
         logger.notice("Hashing cacheable targets")
         let cacheableTargets = try self.cacheableTargets(graph: graph)
 
-        // Build
         logger.notice("Building cacheable targets")
-
-        try cacheableTargets.sorted(by: { $0.key.target.name < $1.key.target.name }).forEach { target, hash in
-            try self.buildAndCacheFramework(path: path, target: target, hash: hash)
+        let sortedCacheableTargets = sortTopologically(cacheableTargets)
+
+        for (index, (target, hash)) in sortedCacheableTargets.enumerated() {
+            // Focus
+            logger.notice("Focusing cacheable targets: \(target.name)")
+            _ = try FocusServiceProjectGeneratorFactory()
+                .generator(sources: [target.name], xcframeworks: false, ignoreCache: false)
+                .generate(path: path, projectOnly: false)
+            
+            // Build
+            logger.notice("Building cacheable targets: \(target.name), \(index + 1) out of \(sortedCacheableTargets.count)")
+            try self.buildAndCacheFramework(path: projectPath, target: target, hash: hash)
         }
 
         logger.notice("All cacheable targets have been cached successfully as \(artifactBuilder.cacheOutputType.description)s", metadata: .success)
     }
+    
+    func sortTopologically(_ cacheableTargets: [TargetNode: String]) -> [(TargetNode, String)] {
```

Comment:
have you considered using [TSCBasic](https://github.com/apple/swift-tools-support-core/blob/32999390a038febc34a412e1f7db5c5456720918/Sources/TSCBasic/GraphAlgorithms.swift#L58) which is a framework we already use extensively?

Reply:
Oh! Nice one, I missed that, I will see if that could be easily reused, will let you know with the findings!